openmdao/components/meta_model_unstructured_comp.py

```diff
@@ -218,12 +221,48 @@ def _setup_partials(self, recurse=True):
                     cols = np.tile(cols, vec_size) + np.repeat(np.arange(vec_size), nnz) * n_wrt
 
                     self._declare_partials(of=of, wrt=wrt, rows=rows, cols=cols)
-
         else:
             # Dense specification of partials for non-vectorized models.
             self._declare_partials(of=[name[0] for name in self._surrogate_output_names],
                                    wrt=[name[0] for name in self._surrogate_input_names])
 
+            # warn the user that if they don’t explicitly set options for fd,
+            #   the defaults will be used
+            # get a list of approximated partials
+            declared_partials = set()
+            for of, wrt, method, fd_options in self._approximated_partials:
+                pattern_matches = self._find_partial_matches(of, wrt)
+                for of_bundle, wrt_bundle in product(*pattern_matches):
+                    of_pattern, of_matches = of_bundle
+                    wrt_pattern, wrt_matches = wrt_bundle
+                    for rel_key in product(of_matches, wrt_matches):
+                        abs_key = rel_key2abs_key(self, rel_key)
+                        declared_partials.add(abs_key)
+            non_declared_partials = []
+            for of, n_of in self._surrogate_output_names:
+                for wrt, n_wrt in self._surrogate_input_names:
+                    abs_key = rel_key2abs_key(self, (of, wrt))
+                    if abs_key not in declared_partials:
+                        non_declared_partials.append((abs_key))
+            if non_declared_partials:
+                msg = "Because the MetaModelUnStructuredComp '{}' uses a surrogate " \
+                       "which does not define a linearize method,\nOpenMDAO will use " \
+                       "finite differences to compute derivates. Some of the derivatives " \
+                       "will be computed\nusing default finite difference " \
+                       "options because they were not explicitly declared.\n".format(self.name)
+                msg += "The derivatives computed using the defaults are:\n"
+                for abs_key in non_declared_partials:
+                    msg += "    {}, {}\n".format(*abs_key)
+                simple_warning(msg, RuntimeWarning)
+
+            for out_name, out_shape in self._surrogate_output_names:
+                surrogate = self._metadata(out_name).get('surrogate')
+                if surrogate and not overrides_method('linearize', surrogate, SurrogateModel):
+                    self._approx_partials(of=out_name,
```

Comment:
If the user declares a partial, and it is in the set of declared_partials, doesn't this code re-declare it by calling self._approx_partials on it? I think you want to exclude the surrogates for which partials have already been declared by checking against `declared_partials` or `non_declared_partials`.   To do this, you will need a test with a really tiny input (so something like f(x) - 1e-9 * sin( 1e-9 * x)), for which the default value will do a horrible job. Then in the test, specify a size that will work for this, like 1e-14.  This test should fail if the default size overrides the requested size. That way you will be certain.

Reply:
edited the equation for the poor fd sine

Reply:
Test is actually covered, and fd seems to address my concern natively.